packages/expo-network/android/src/main/java/expo/modules/network/NetworkModule.kt

```diff
@@ -0,0 +1,158 @@
+    else NetworkStateType.UNKNOWN
+
+  private fun rawIpToString(ip: Int): String {
+    // Convert little-endian to big-endian if needed
+    val ip = if (ByteOrder.nativeOrder() == ByteOrder.LITTLE_ENDIAN) {
+      Integer.reverseBytes(ip)
+    } else {
+      ip
+    }
+
+    var ipByteArray = BigInteger.valueOf(ip.toLong()).toByteArray()
+    if (ipByteArray.size < 4) {
+      ipByteArray = frontPadWithZeros(ipByteArray)
+    }
+
+    return try {
+      InetAddress.getByAddress(ipByteArray).hostAddress
+    } catch (e: UnknownHostException) {
+      "0.0.0.0"
+    }
+  }
+
+  @ExpoMethod
+  fun getNetworkStateAsync(promise: Promise) {
+    val result = Bundle()
+    val connectivityManager = appContext.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+
+    try {
+      if (Build.VERSION.SDK_INT < 29) { // use getActiveNetworkInfo before api level 29
+        val netInfo = connectivityManager.activeNetworkInfo
+        val connectionType = getConnectionType(netInfo)
+
+        result.apply {
+          putBoolean("isInternetReachable", netInfo!!.isConnected)
+          putString("type", connectionType.value)
+          putBoolean("isConnected", connectionType.isDefined)
+        }
+
+        promise.resolve(result)
+      } else {
+        val network = connectivityManager.activeNetwork
+        val isInternetReachable = network != null
+
+        var connectionType: NetworkStateType? = if (isInternetReachable) {
+          val netCapabilities = connectivityManager.getNetworkCapabilities(network)
+          getConnectionType(netCapabilities)
+        } else {
+          null
+        }.also { connectionType ->
+          result.apply {
+            putString("type", connectionType?.value ?: NetworkStateType.NONE.value)
+            putBoolean("isInternetReachable", isInternetReachable)
+            putBoolean("isConnected", connectionType != null && connectionType.isDefined)
+          }
+        }
+        promise.resolve(result)
+      }
+    } catch (e: Exception) {
+      promise.reject("ERR_NETWORK_NO_ACCESS_NETWORKINFO", "Unable to access network information", e)
+    }
+  }
+
+  @ExpoMethod
+  fun getIpAddressAsync(promise: Promise) {
+    try {
+      promise.resolve(rawIpToString(wifiInfo.ipAddress))
+    } catch (e: Exception) {
+      Log.e(TAG, e.message ?: "Could not get ip address")
+      promise.reject("ERR_NETWORK_IP_ADDRESS", "Unknown Host Exception", e)
+    }
+  }
+
+  @ExpoMethod
+  fun isAirplaneModeEnabledAsync(promise: Promise) {
+    val isAirPlaneMode = Settings.Global.getInt(appContext.contentResolver, Settings.Global.AIRPLANE_MODE_ON, 0) != 0
+    promise.resolve(isAirPlaneMode)
+  }
+
+  companion object {
+    private fun frontPadWithZeros(inputArray: ByteArray): ByteArray {
+      val newByteArray = byteArrayOf(0, 0, 0, 0)
+      System.arraycopy(inputArray, 0, newByteArray, 4 - inputArray.size, inputArray.size)
+      return newByteArray
+    }
```

Comment:
Move this into module scope since it's private and we don't need a companion object.